pydantic/_internal/_discriminated_union.py

```diff
@@ -12,6 +12,28 @@
 def apply_discriminator(
     schema: core_schema.CoreSchema, discriminator: str, definitions: dict[str, core_schema.CoreSchema] | None = None
 ) -> core_schema.CoreSchema:
+    """Applies the discriminator and returns a new core schema.
+
+    Args:
+        schema: The input schema.
+        discriminator: The name of the field which will serve as the discriminator.
+        definitions: A mapping of schema ref to schema.
+
+    Returns:
+        The new core schema.
+
+    Raises:
+        TypeError:
+            * If `discriminator` is used with invalid union variant.
+            * If `discriminator` is used with `Union` type with one variant.
+            * If `discriminator` value mapped to multiple choices.
+        ValueError: If the definition for ref is missing.
```

Comment:
```suggestion
        ValueError:
            If the definition for ref is missing.
```
this may help it render better in the API docs, responding to Marcelo's point above. I haven't checked — maybe it's fine now, or maybe this doesn't actually help. But I think it at least makes it easier to read in the source code 😄